src/main/java/chessmaster/game/Game.java

```diff
@@ -96,6 +97,11 @@ public void run() {
                 ui.printErrorMessage(e);
             }
         }
+        if (hasEnded || RestartCommand.isRestartCommand(command)) {
+            return true;
+        } else {
+            return false;
+        }
```

Comment:
Just return directly to keep the logic short
`return hasEnded || RestartCommand.isRestart(command);`